sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -6272,21 +6272,21 @@ pi_result piextGetDeviceFunctionPointer(pi_device Device, pi_program Program,
                                          0, nullptr, &Size);
     if (ret_err != PI_SUCCESS) {
       *FunctionPointerRet = 0;
-      return ret_err;
+      return PI_FALLBACK_FAILURE;
     }
 
     std::string ClResult(Size, ' ');
     ret_err = piProgramGetInfo(Program, PI_PROGRAM_INFO_KERNEL_NAMES,
                                ClResult.size(), &ClResult[0], nullptr);
     if (ret_err != PI_SUCCESS) {
       *FunctionPointerRet = 0;
-      return ret_err;
+      return PI_FALLBACK_FAILURE;
     }
 
     // Get rid of the null terminator and search for kernel_name
     ClResult.pop_back();
     *FunctionPointerRet = is_in_separated_string(ClResult, ';', FunctionName);
-    return PI_SUCCESS;
+    return PI_FALLBACK_SUCCESS;
```

Comment:
Why is this unconditionally returning success ? it should depend on the search results, right? 
Also let's not write the boolean value to the returned pointer, just always write 0.

Reply:
So this is because even if a function cannot be found you want to handle it as a success if it is a call from `has_kernel`. I think a reasonable solution could be to remove `PI_FALLBACK_SUCCESS` and `PI_FALLBACK_FAILURE` and instead have `PI_FUNCTION_ADDRESS_IS_NOT_AVAILABLE` and `PI_FUNCTION_NOT_AVAILABLE` to indicate that the function does not exist and let that error code be handled by `has_kernel`.

Reply:
I correct my comment I am not adding `PI_FUNCTION_NOT_AVAILABLE` and am using `PI_INVALID_KERNEL_NAME` to indicate that the function does not exist